tests/lua/type_mismatch.lua

```diff
@@ -38,6 +38,8 @@ test.type_mismatch.unsupported_type_p = function(wrong_arg_num, func_name, ...)
 end
```

Comment:
This file is too sophisticated and hard to read.
Test cases not isolated and it is very hard to understand underlying meaning.

Reply:
Yes, it's a little bit more complex than other tests, but this is a cost for test generation (and usage of `u-test` =) )